src/types/redis_zset.cc

```diff
@@ -744,16 +754,20 @@ rocksdb::Status ZSet::UnionStore(const Slice &dst, const std::vector<KeyWeight>
       }
     }
   }
-  if (!dst_zset.empty()) {
-    std::vector<MemberScore> mscores;
-    mscores.reserve(dst_zset.size());
+  auto compare_score = [](const MemberScore &score1, const MemberScore &score2) {
+    if (score1.score == score2.score) {
+      return score1.member < score2.member;
+    }
+    return score1.score < score2.score;
+  };
+  if (members && !dst_zset.empty()) {
+    members->reserve(dst_zset.size());
     for (const auto &iter : dst_zset) {
-      mscores.emplace_back(MemberScore{iter.first, iter.second});
+      members->emplace_back(MemberScore{iter.first, iter.second});
     }
-    if (saved_cnt) *saved_cnt = mscores.size();
-    Overwrite(dst, mscores);
+    if (saved_cnt) *saved_cnt = members->size();
+    std::sort(members->begin(), members->end(), compare_score);
```

Comment:
Sorry, maybe I misunderstand your previous comment. We don't need to sort members here, because the underlying storage will sort them.

Reply:
Underlying storage will sort them for ZUNIONSTORE, but we don’t use the storage for ZUNION, so i think to keep the response consistent, we’ll need to sort it at this level (atleast for ZUNION) command

Reply:
@uds5501 OK, then it's good for me.  How about moving the sort logic to the ZUNION command since we don't need to sort for the union store?

Reply:
Makes sense @git-hulk.

Reply:
Incorporated this change.